docs/unnecessary-allof-ref-wrapper.md

```diff
@@ -0,0 +1,37 @@
+---
+title: Remove single-\`$ref\` \`allOf\` wrapper
+code: unnecessary-allof-ref-wrapper
+categories: readability, style
+dialects: 2019-09, 2020-12, draft4, draft6, draft7
```

Comment:
The `allOf` wrapper is necessary for draft-04/6/7. The definition of `$ref` changed in 2019-09 that made the wrapper no longer necessary.